chore: update module plugin docs and readme (#3782)
web-infra-dev/modern.js: packages/document/module-doc/docs/en/plugins/official-list/plugin-polyfill.md

> # Polyfill plugin

Reviewer: ```suggestion
# Polyfill Plugin
```

:::tip

> Normally, we don't need to inject polyfill for npm packages, this step should be done on the framework application side, but in some scenarios we need to inject polyfill in order to make our library run directly in low version browsers.

Reviewer: ```suggestion
Normally, we don't need to inject polyfills for npm packages. This step should be completed on the framework side of web applications. However, in some scenarios, we need to inject polyfills in order to make our library run directly on legacy browsers.
```

> Note that this plugin does not transform your code syntax, it only injects polyfill for unsupported functions used in your code, importing them as normal functions instead of polluting the global. You need to install the core-js-pure dependency

Reviewer: ```suggestion
Note that this plugin does not transform your code syntax, it only injects polyfill for unsupported functions used in your code, importing them as normal functions instead of polluting the global. You need to install the `core-js-pure` dependency.
```

:::

## Quick start

### Install

```bash
# npm
npm install @modern-js/plugin-module-polyfill -D

# yarn
yarn add @modern-js/plugin-module-polyfill -D

# pnpm
pnpm add @modern-js/plugin-module-polyfill -D
```

### Register

In Module Tools, you can register plugins in the following way:

```ts
import moduleTools, { defineConfig } from '@modern-js/module-tools';
import { modulePluginPolyfill } from '@modern-js/plugin-module-polyfill';

export default defineConfig({
  plugins: [
    moduleTools(),
    modulePluginPolyfill({
      targets: "> 0.25%, not dead"
    }),
  ],
});
```

## Config

> * type:

Reviewer: ```suggestion
* Type:
```
